Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkBasic/PsychEyelinkDispatchCallback.m

```diff
@@ -129,326 +161,419 @@
+                    Screen(offscreen,'TextSize',el.imgtitlefontsize);
+                    Screen('DrawText', offscreen, imgtitle, width/2-dw/2, heigth/2+dh/2+h2, el.imgtitlecolour);
+                    Screen('SelectStereoDrawBuffer', eyewin, it); % select left eye window
+                    Screen('DrawTexture',eyewin,offscreen,  [width/2-dw/2 heigth/2+dh/2+h2 width/2-dw/2+500 heigth/2+dh/2+h2+500], [width/2-dw/2 heigth/2+dh/2+h2 width/2-dw/2+500 heigth/2+dh/2+h2+500]);
+                    Screen('Close',offscreen);
+                    
+                    lasttitle = imgtitle;
+                end
+            catch %myerr
+                fprintf('EyelinkDrawCameraImage:error \n');
+                disp(psychlasterror);
             end
+        end
+    end
+
+
+
 
-            sn = Screen('WindowScreenNumber', eyewin);
-            offscreen = Screen('OpenOffscreenWindow', sn, el.backgroundcolour, [], [], 32);
-            Screen(offscreen,'TextFont',el.imgtitlefont);
-            Screen(offscreen,'TextSize',el.imgtitlefontsize);
-            Screen('DrawText', offscreen, imgtitle, width/2-dw/2, heigth/2+dh/2+h2, el.imgtitlecolour);
-            Screen('DrawTexture',eyewin,offscreen,  [width/2-dw/2 heigth/2+dh/2+h2 width/2-dw/2+500 heigth/2+dh/2+h2+500], [width/2-dw/2 heigth/2+dh/2+h2 width/2-dw/2+500 heigth/2+dh/2+h2+500]);
-            Screen('Close',offscreen);
-            newImage = 0;
+
+    function EyelinkDrawCalibrationTarget(eyewin, el, calxy)
+        if el.winInfo.StereoMode ~= 0
+            drawScreens = 2; % stereoscopic drawing
+        else
+            drawScreens = 1; % non-stereoscopic drawing
+        end
+
+        [width, heigth]=Screen('WindowSize', eyewin);
+
+        for it = 0:drawScreens-1
+            Screen('SelectStereoDrawBuffer', eyewin, it); % select eye window
+            switch el.calTargetType
+                case 'video'
+                    if( ~isempty(el.calAnimationTargetFilename) && ~isempty(eyelinkanimationtarget))
+                        rect=CenterRectOnPoint([0 0 eyelinkanimationtarget.imgw eyelinkanimationtarget.imgh], calxy(1), calxy(2));
+                        if it == 0
+                            tex=Screen('GetMovieImage', eyewin, eyelinkanimationtarget.movie, el.calAnimationWaitTex);
+                        end
+                        if(tex>0)
+                            Screen('DrawTexture', eyewin , tex, [], rect, [], 0);
+                            if drawScreens == 1 || (drawScreens == 2 && it == 1)
+                                Screen('Flip', eyewin);
+                            end
+                        end
+                        if it == drawScreens-1 && tex > 0
+                            Screen('Close', tex);
+                        end
+                    end
+                    
+                case 'image'
+                    if ~isempty(el.calImageInfo) && ~isempty(el.calImageTexture)
```

Comment:
el.calImageTexture can not be empty, or the following DrawTexture call would error out.
Not sure if this if statement makes any sense at all, as if the rect=CenterRect... code would be skipped, then I think the variable rect would not exist, and DrawTexture would error out. So if this assignment is always needed, then the if statement can be dropped.